test/y2partitioner/sequences/controllers/lvm_lv_test.rb

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../../test_helper"
+require "y2partitioner/device_graphs"
+require "y2partitioner/sequences/controllers/lvm_lv"
+
+describe Y2Partitioner::Sequences::Controllers::LvmLv do
+  using Y2Storage::Refinements::SizeCasts
+
+  subject(:controller) { described_class.new(vg) }
+
+  let(:current_graph) { Y2Partitioner::DeviceGraphs.instance.current }
+
+  let(:vg) { Y2Storage::LvmVg.find_by_vg_name(current_graph, vg_name) }
+
+  let(:vg_name) { "vg0" }
+
+  before do
+    devicegraph_stub("lvm-two-vgs.yml")
+
+    allow(controller).to receive(:lv_name).and_return(lv_name)
+    allow(controller).to receive(:size).and_return(size)
+    allow(controller).to receive(:stripes_number).and_return(stripes_number)
+    allow(controller).to receive(:stripes_size).and_return(stripes_size)
+  end
+
+  let(:lv_name) { nil }
+
+  let(:size) { nil }
+
+  let(:stripes_number) { nil }
+
+  let(:stripes_size) { nil }
+
+  describe "#vg" do
+    it "returns a Y2Storage::LvmVg" do
+      expect(subject.vg).to be_a(Y2Storage::LvmVg)
+    end
+
+    it "returns the currently editing vg" do
+      expect(subject.vg).to eq(vg)
+    end
+  end
+
+  describe "#create_lv" do
+    let(:lv_name) { "lv3" }
+
+    let(:size) { 3.GiB }
+
+    let(:stripes_number) { 2 }
+
+    let(:stripes_size) { 16.KiB }
+
+    it "creates a new lv in the editing vg" do
+      expect(controller.vg.lvm_lvs.size).to eq(2)
+      controller.create_lv
+      expect(controller.vg.lvm_lvs.size).to eq(3)
+    end
+
+    it "creates a new lv with indicated values" do
+      expect(controller.vg.lvm_lvs.map(&:lv_name)).to_not include(lv_name)
+
+      controller.create_lv
+      lv = controller.vg.lvm_lvs.find { |l| l.lv_name == lv_name }
+
+      expect(lv.lv_name).to eq(lv_name)
+      expect(lv.size).to eq(size)
+      expect(lv.stripes).to eq(stripes_number)
+      expect(lv.stripe_size).to eq(stripes_size)
+    end
+
+    it "stores the new created lv" do
+      expect(controller.lv).to be_nil
+
+      controller.create_lv
+
+      expect(controller.lv).to_not be_nil
```

Comment:
np: I think the test below makes this one unneeded, but it is true that error message could be slightly better.